packages/plugin-commands-patching/test/patch.test.ts

```diff
@@ -93,4 +94,31 @@ describe('patch and commit', () => {
     await expect(() => patch.handler({ ...defaultPatchOption, editDir }, ['is-positive@1.0.0']))
       .rejects.toThrow(`The target directory already exists: '${editDir}'`)
   })
+
+  test('should work when there is a no eol error', async () => {
+    const output = await patch.handler(defaultPatchOption, ['safe-execa@0.1.2'])
+    const userPatchDir = output.substring(output.indexOf(':') + 1).trim()
+    const tempDir = os.tmpdir()
+
+    expect(userPatchDir).toContain(tempDir)
+    expect(fs.existsSync(userPatchDir)).toBe(true)
+    expect(fs.existsSync(path.join(userPatchDir, 'lib/index.js'))).toBe(true)
+
+    fs.appendFileSync(path.join(userPatchDir, 'lib/index.js'), '\n// test patching', 'utf8')
+
+    await patchCommit.handler({
+      ...DEFAULT_OPTS,
+      dir: process.cwd(),
+    }, [userPatchDir])
+
+    const { manifest } = await readProjectManifest(process.cwd())
+    expect(manifest.pnpm?.patchedDependencies).toStrictEqual({
+      'safe-execa@0.1.2': 'patches/safe-execa@0.1.2.patch',
+    })
+    const patchContent = fs.readFileSync('patches/safe-execa@0.1.2.patch', 'utf8')
+    expect(patchContent).toContain('diff --git')
+    expect(patchContent).toContain('// test patching')
+    expect(patchContent).toContain('No newline at end of file.')
```

Comment:
@await-ovo as you can see I have added a check for whether the patch contains that string. And it doesn't.

Reply:
@zkochan  Yeah, I replaced the `No newline at end of file.`  in the patchContent with an empty string in the commit , so the patch ended up working fine.

After removing this line(https://github.com/await-ovo/pnpm/blob/fix-patch-commit-no-eol/packages/plugin-commands-patching/src/patchCommit.ts#L101), you can see the `No newline at end of file.` in the patchContent, and you can reproduce the problem at the beginning ~